### PR TITLE
feat: Nmp 36 docker update backend

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

`docker compose up --watch` now runs API in a container with hot reloading. 

## Type of changes

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-62-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-62-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by LocalNewsTV** [comment]
View the comments on https://github.com/bcgov/nr-sustainment-capstone-2024/pull/41 for whats related to your issues

File: backend/src/controllers/health-api-controller.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+ * @desc Simple health endpoint to check API status
+ * @author GDamaso
+ */
 import { Response, Request } from 'express';
 
-// If we're healthy...
+
 const checkHealth = (req: Request, res: Response) => {
-  res.status(200).send('NMP API is healthy and ready!');
+  res.status(200).send('NMP API is healthy and ready! Hot reloading all the way!');
```
Comment (LocalNewsTV) [inline backend/src/controllers/health-api-controller.ts:9]:
```suggestion
  res.status(200).send('NMP API is healthy and ready!');
```
This response is used for your health check

File: docker-compose.yml [hunk 2]
```diff
@@ -34,10 +34,29 @@ services:
...
+      - 3000:3000
+    develop:
+      watch:
+        - action: sync
+          path: ./backend/src
+          target: /backend/src
+          ignore:
+            - node_modules
+            - .github/
+            - .git/
```
Comment (LocalNewsTV) [inline docker-compose.yml:59]: - Container should depend on db being healthy
- Container should health check with `curl localhost:${PORT:-3000}/api/health | grep 'NMP API is healthy and ready!'`

**Comment by LocalNewsTV** [comment]
Also, there is an error when I attempt to update a file using this 
backend/src/controllers/health-api-controller.ts: Cannot open: File exists
tar: Exiting with failure status due to previous errors
time="2024-05-14T16:25:11-07:00" level=warning msg="Error handling changed files for service backend: 1 error occurred:\n\t* copying files to c2f686decb8045b076b929c6e323e7dbf5b3f57f48145aeb825e6ee787031694: exit code 2\n\n"

File: backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# copy everything from current|root dir to WORKDIR
+COPY . .
+
+# Install all dependencies and typescript
+RUN npm i 
+RUN npm i typescript
+
+# Build the API, target is /backend/dist
+RUN npm run build
```
Comment (LocalNewsTV) [inline backend/Dockerfile:15]: You don't need to run build since you have this running dev

**Comment by GDamaso** [comment]
@LocalNewsTV I haven't seen that error before. 
Do you have any more information about it?

**Review by LocalNewsTV — APPROVED** [review]
Apparently my Docker desktop was out of date, triggering the error
Some changes were made to Docker compose watch that weren't compatible with my version triggering the error when files change. (Even though I was using `docker compose watch` a month ago for personal projects` 
Looks good, runs as expected health checks functioning as anticipated

**Pull request merged**